.github/workflows/ci.yml

```
@@ -84,38 +77,23 @@ jobs:
-        uv run ruff format .
-    - name: Verify Builder
-      run: |
-        uv run python build.py
-    - name: Commit and Push changes
-      env:
-        REF_NAME: ${{ github.head_ref || github.ref_name }}
+      run: uv sync --all-extras --frozen --no-compile
+    - name: Run Final CI Gate
+      run: uv run python tools/pre_ci.py "${{ env.MIN_VER }}" "${{ env.MAX_VER }}"
+    - name: Commit and Push
       run: |
```

actionlint (1.7.12)

[error] "github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)